packages/plugin-npm/sources/index.ts

```diff
@@ -43,6 +43,31 @@ const registrySettings: {[name: string]: SettingsDefinition} = {
   },
 };
 
+declare module '@yarnpkg/core' {
+  interface ConfigurationValueMap {
+    npmAlwaysAuth: boolean;
+    npmAuthIdent: string|null;
+    npmAuthToken: string|null;
+
+    npmPublishRegistry: string | null;
+    npmRegistryServer: string;
+
+    npmScopes: Map<string, {
+      get(key: `npmAlwaysAuth`): boolean;
+      get(key: `npmAuthIdent`): string|null;
+      get(key: `npmAuthToken`): string|null;
+
+      get(key: `npmPublishRegistry`): string | null;
+      get(key: `npmRegistryServer`): string;
+    }>;
```

Comment:
Would it make sense to abstract that in an utility? Something like:

```ts
type ConfigurationPick<T> = Map<string, Pick<ConfigurationValueMap, T>>;
declare const npmScopes: Map<string, ConfigurationPick<'npmAlwaysAuth' | 'npmAuthIdent' | ...>>
```

Reply:
I've made a slightly different utility that's not limited to our current `npmScopes` and `npmRegistries` configuration types, which are indeed subsets of the entire configuration.